src/System.Drawing.Common/src/System/Drawing/Drawing2D/GraphicsPathIterator.cs

```diff
@@ -169,73 +168,51 @@ public unsafe int Enumerate(ref PointF[] points, ref byte[] types)
             if (points.Length != types.Length)
                 throw SafeNativeMethods.Gdip.StatusException(SafeNativeMethods.Gdip.InvalidParameter);
 
-            int resultCount = 0;
-            int size = Marshal.SizeOf(typeof(GPPOINTF));
-            int count = points.Length;
-            byte[] typesLocal = new byte[count];
+            if (points.Length == 0)
+                return 0;
 
-            IntPtr memoryPts = Marshal.AllocHGlobal(checked(count * size));
-            try
+            fixed (PointF* p = points)
+            fixed (byte* t = types)
             {
-                int status = SafeNativeMethods.Gdip.GdipPathIterEnumerate(new HandleRef(this, nativeIter), out resultCount,
```

Comment:
Would be nice to rename nativeIter to nativeIterator, or something like this, perhaps not in the scope of this change.

Reply:
I'm going to follow up this change with a style pass. I'll look at private/internal members.